### PR TITLE
feat: add GITHUB_REF_NAME env var

### LINKED ISSUE #71: Provide a shared env variable between "pull_request" and "push" events for the branch name
**Is your feature request related to a problem? Please describe.**
The value of `GITHUB_REF_SLUG` is different depending on whether a pull_request/push event is the origin.

It makes it harder to write scripts where we don't really care what's the event, we only want the branch name.

```
Run rlespinasse/github-slug-action@v3.x
##[debug]Node Action run completed with exit code 0
##[debug]GITHUB_EVENT_PULL_REQUEST_HEAD_SHA_SHORT='8c2d7bad'
##[debug]GITHUB_REPOSITORY_OWNER_PART='UnlyEd'
##[debug]GITHUB_REPOSITORY_NAME_PART='github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_SLUG='unlyed-github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_SLUG_CS='UnlyEd-github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_OWNER_PART_SLUG='unlyed'
##[debug]GITHUB_REPOSITORY_OWNER_PART_SLUG_CS='UnlyEd'
##[debug]GITHUB_REPOSITORY_NAME_PART_SLUG='github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_NAME_PART_SLUG_CS='github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_SLUG_URL='unlyed-github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_SLUG_URL_CS='UnlyEd-github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_OWNER_PART_SLUG_URL='unlyed'
##[debug]GITHUB_REPOSITORY_OWNER_PART_SLUG_URL_CS='UnlyEd'
##[debug]GITHUB_REPOSITORY_NAME_PART_SLUG_URL='github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_NAME_PART_SLUG_URL_CS='github-action-deploy-on-vercel'
##[debug]GITHUB_REF_SLUG='18-merge'
##[debug]GITHUB_REF_SLUG_CS='18-merge'
##[debug]GITHUB_HEAD_REF_SLUG='feature-dynamic-aliases'
##[debug]GITHUB_HEAD_REF_SLUG_CS='feature-dynamic-aliases'
##[debug]GITHUB_BASE_REF_SLUG='main'
##[debug]GITHUB_BASE_REF_SLUG_CS='main'
##[debug]GITHUB_REF_SLUG_URL='18-merge'
##[debug]GITHUB_REF_SLUG_URL_CS='18-merge'
##[debug]GITHUB_HEAD_REF_SLUG_URL='feature-dynamic-aliases'
##[debug]GITHUB_HEAD_REF_SLUG_URL_CS='feature-dynamic-aliases'
##[debug]GITHUB_BASE_REF_SLUG_URL='main'
##[debug]GITHUB_BASE_REF_SLUG_URL_CS='main'
##[debug]GITHUB_SHA_SHORT='b97c3a2a'
##[debug]Finishing: Expose GitHub slug/short variables
``` 

```
Run rlespinasse/github-slug-action@v3.x
##[debug]Node Action run completed with exit code 0
##[debug]GITHUB_EVENT_REF_SLUG='feature-dynamic-aliases'
##[debug]GITHUB_EVENT_REF_SLUG_CS='feature-dynamic-aliases'
##[debug]GITHUB_EVENT_REF_SLUG_URL='feature-dynamic-aliases'
##[debug]GITHUB_EVENT_REF_SLUG_URL_CS='feature-dynamic-aliases'
##[debug]GITHUB_REPOSITORY_OWNER_PART='UnlyEd'
##[debug]GITHUB_REPOSITORY_NAME_PART='github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_SLUG='unlyed-github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_SLUG_CS='UnlyEd-github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_OWNER_PART_SLUG='unlyed'
##[debug]GITHUB_REPOSITORY_OWNER_PART_SLUG_CS='UnlyEd'
##[debug]GITHUB_REPOSITORY_NAME_PART_SLUG='github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_NAME_PART_SLUG_CS='github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_SLUG_URL='unlyed-github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_SLUG_URL_CS='UnlyEd-github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_OWNER_PART_SLUG_URL='unlyed'
##[debug]GITHUB_REPOSITORY_OWNER_PART_SLUG_URL_CS='UnlyEd'
##[debug]GITHUB_REPOSITORY_NAME_PART_SLUG_URL='github-action-deploy-on-vercel'
##[debug]GITHUB_REPOSITORY_NAME_PART_SLUG_URL_CS='github-action-deploy-on-vercel'
##[debug]GITHUB_REF_SLUG='feature-dynamic-aliases'
##[debug]GITHUB_REF_SLUG_CS='feature-dynamic-aliases'
##[debug]GITHUB_REF_SLUG_URL='feature-dynamic-aliases'
##[debug]GITHUB_REF_SLUG_URL_CS='feature-dynamic-aliases'
##[debug]GITHUB_SHA_SHORT='8c2d7bad'
##[debug]Finishing: Expose GitHub slug/short variables
```

We can see in the first case we should use the `GITHUB_HEAD_REF_SLUG` and in the second case it should be `GITHUB_REF_SLUG`. 

**Describe the solution you'd like**
Adding a `GITHUB_BRANCH_NAME` env variable that is contains the branch name in both cases.

**Describe alternatives you've considered**
Writing our own custom if/else in github action, but that could be avoided.

### PR DESCRIPTION
**Pull request by Ameausoone**
Resolves #71

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -334,22 +353,51 @@ function exportSecondPartSlugUrl(
...
+  const isPullRequest = !!process.env.GITHUB_HEAD_REF
+  let branchName
+  core.info(`isPullRequest : ${isPullRequest}`)
+  if (isPullRequest) {
+    branchName = process.env.GITHUB_HEAD_REF
+    core.info(`branchName : ${branchName}`)
+  } else if (process.env.GITHUB_REF) {
+    branchName = process.env.GITHUB_REF
+  }
+  core.info(`branchName : ${branchName}`)
```
Comment (rlespinasse) [inline src/main.ts]:
```suggestion
  if (isPullRequest) {
    branchName = process.env.GITHUB_HEAD_REF
  } else if (process.env.GITHUB_REF) {
    branchName = process.env.GITHUB_REF
  }
  core.info(`branchName : ${branchName}`)
```

**Review by rlespinasse — COMMENTED** [review]
Can you add some documentation on the README?

**Comment by rlespinasse** [comment]
@Ameausoone Can you also cleanup the commit messages?

File: src/main.ts [hunk 2]
```diff
@@ -81,6 +91,7 @@ const GITHUB_EVENT_PULL_REQUEST_HEAD_SHA_SHORT =
   'GITHUB_EVENT_PULL_REQUEST_HEAD_SHA_SHORT'
 
 async function run(): Promise<void> {
+  core.exportVariable('GITHUB_YOLO', 'yolo')
```
Comment (rlespinasse) [inline src/main.ts]: YOLO 😱

File: src/main.ts [hunk 3]
```diff
@@ -334,22 +355,53 @@ function exportSecondPartSlugUrl(
...
+function exportBranchName(): void {
+  //GITHUB_HEAD_REF is only set for pull request events https://docs.github.com/en/actions/reference/environment-variables
+  const isPullRequest = !!process.env.GITHUB_HEAD_REF
+  let branchName
+  console.log(`process.env.GITHUB_HEAD_REF : ${process.env.GITHUB_HEAD_REF}`)
+  console.log(`isPullRequest : ${isPullRequest}`)
+  console.log(`GITHUB_REF : ${GITHUB_REF}`)
+  if (isPullRequest) {
+    branchName = process.env.GITHUB_HEAD_REF
+    core.info(`branchName : ${branchName}`)
```
Comment (rlespinasse) [inline src/main.ts]: remove this line. the branchName is printed later on

**Comment by rlespinasse** [comment]
@Ameausoone can you review my additional changes to this PR?

**Review by rlespinasse — APPROVED** [review]

**Comment by rlespinasse** [comment]
Since **GITHUB_REF** can also contain a tag name, I think we have two choices

1. Use **REF** (instead of `BRANCH`) like `GITHUB_REF_NAME` (or `GITHUB_REF_LABEL`)
2. Use two variables `GITHUB_BRANCH_NAME`, and `GITHUB_TAG_NAME` (set when appropriate).

@Ameausoone any thoughts on that matter?

File: src/main.ts [hunk 4]
```diff
@@ -334,22 +353,48 @@ function exportSecondPartSlugUrl(
...
+  //GITHUB_HEAD_REF is only set for pull request events https://docs.github.com/en/actions/reference/environment-variables
+  const isPullRequest = !!process.env.GITHUB_HEAD_REF
+  let branchName
+  if (isPullRequest) {
+    branchName = process.env.GITHUB_HEAD_REF
+  } else {
+    branchName = process.env.GITHUB_REF
+  }
+  if (branchName) {
+    core.exportVariable(GITHUB_BRANCH_NAME, branchName)
```
Comment (rlespinasse) [inline src/main.ts]:
```suggestion
    core.exportVariable(GITHUB_BRANCH_NAME, removeRef(branchName))
```

**Review by rlespinasse — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 4 above)
Comment (Ameausoone) [reply src/main.ts]: 👍

**Comment by Ameausoone** [comment]
> Since **GITHUB_REF** can also contain a tag name, I think we have two choices
> 
> 1. Use **REF** (instead of `BRANCH`) like `GITHUB_REF_NAME` (or `GITHUB_REF_LABEL`)
> 2. Use two variables `GITHUB_BRANCH_NAME`, and `GITHUB_TAG_NAME` (set when appropriate).
> 
> @Ameausoone any thoughts on that matter?

Good idea, I vote for the 1st solution

File: .github/workflows/release-dryrun.yml [hunk 5]
```diff
@@ -18,7 +18,7 @@ jobs:
         with:
           branches: |
             [
-              "${{ env.GITHUB_REF_SLUG }}"
+              "${{ env.GITHUB_BRANCH_NAME }}"
```
Comment (rlespinasse) [inline .github/workflows/release-dryrun.yml]:
```suggestion
              "${{ env.GITHUB_REF_NAME }}"
```

File: README.md [hunk 6]
```diff
@@ -66,6 +68,14 @@ Check for more [examples][examples] (OS usage, URL use, ...)
 
 **Note**: If you don't find what you search for, read more about [available `GitHub` variables](docs/github-variables.md), and propose a [new custom variable][custom-variable].
 
+### Enhanced variables
+
+- `GITHUB_BRANCH_NAME` will contains the branch name
```
Comment (rlespinasse) [inline README.md]:
```suggestion
- `GITHUB_REF_NAME` will contains the reference name (branch or tag)
```

**Review by rlespinasse — APPROVED** [review]

**Pull request merged**